tests/50federation/40publicroomlist.pl

```diff
@@ -0,0 +1,104 @@
+# Copied from 30rooms/70publicroomslist.pl, modified to test the federation publicRooms API
+test "Name/topic keys are correct",
+   requires => [ local_user_fixture(), $main::HOMESERVER_INFO[0], $main::HTTP_CLIENT ],
+
+   check => sub {
+      my ( $user, $info, $client ) = @_;
+
+      my $server_name = $info->server_name;
+
+      my %rooms = (
+         publicroomalias_no_name => {},
+         publicroomalias_with_name => {
+            name => "name_1",
+         },
+         publicroomalias_with_topic => {
+            topic => "topic_1",
+         },
+         publicroomalias_with_name_topic => {
+            name => "name_2",
+            topic => "topic_2",
+         },
+      );
+
+      Future->needs_all( map {
+         my $alias_local = $_;
+         my $room = $rooms{$alias_local};
+
+         matrix_create_room( $user,
+            visibility      => "public",
+            room_alias_name => $alias_local,
+            name            => $room->{name},
+            topic           => $room->{topic},
+         )
+      } keys %rooms )
+      ->then( sub {
+         $client->do_request_json(
+            method => "GET",
+            uri    => "/r0/publicRooms",
+            uri => "https://$server_name/_matrix/federation/v1/publicRooms",
```

Comment:
You've specified `uri` twice; only the latter will take effect. Maybe line 38 just wants deleting?